### ISSUE TITLE
defaultInterpretOptions is not inherited from parent record

### ISSUE THREAD

**Issue by PierreR** [Gabriel439/Haskell-Dhall-Library#33]
Given this code:
```
auto :: (GenericInterpret (Rep a), Generic a) => Type a
auto = deriveAuto
  ( defaultInterpretOptions { fieldModifier = Data.Text.Lazy.dropWhile (== '_') })

data GitRepo
  = GitRepo
  { _host :: LText
  , _repo :: LText
  } deriving (Generic, Show)

data BoxConfig
  = BoxConfig
  { _userName        :: LText
  , _dotfilesRepo    :: GitRepo
  } deriving (Generic, Show)

makeLenses ''BoxConfig
makeLenses ''GitRepo

instance Interpret GitRepo
instance Interpret BoxConfig

boxConfig :: IO BoxConfig
boxConfig = Dhall.input auto "/vagrant/config/box"
```

It will force me to have this config file:
```
{ userName       = "John Doe"
, dotfilesRepo   = { _host = "github", _repo = "jdoe/dotfiles.git"}
}
```
but I wish to have
```
{ userName       = "John Doe"
, dotfilesRepo   = { host = "github", repo = "jdoe/dotfiles.git"}
}
```

**Comment by PierreR** [comment]
@Gabriel439 Thanks for the quick fix ! Works like a charm.

**Comment by PierreR** [comment]
@Gabriel439 I am now using a list and it doesn't bubble down ...
```
{ userName       = "John Doe"
, additionalRepos = [ { checkout = "vcsh clone git://github.com/PierreR/dotfiles.git",
                        push = "vcsh dotfiles push git@mygithub.com:PierreR/dotfiles.git" }
                    ]                            ]
}
```
```
data MrRepo
  = MrRepo
  { _checkout :: LText
  , _push     :: LText
  } deriving (Generic, Show)

data BoxConfig
  = BoxConfig
  { _userName        :: LText
  , _additionalRepos :: Vector MrRepo
  } deriving (Generic, Show)
```
```
Error: Expression doesn't match annotation
/vagrant/config/box : { additionalRepos : List { _checkout : Text, _push : Text }, userName : Text }
```

**Comment by Gabriel439** [comment]
Latter issue fixed by e7e799f4daf85fb4a916af4eec6a142589b66892

**Comment by Gabriel439** [comment]
You're welcome!
